src/main/java/org/jboss/tm/usertx/TransactionListenerRegistry.java

```diff
@@ -0,0 +1,15 @@
+package org.jboss.tm.usertx;
+
+import javax.transaction.NotSupportedException;
+import javax.transaction.Transaction;
+
+public interface TransactionListenerRegistry {
+    /**
+     * @param transaction the transaction that the caller is interested in receiving events for (must not be null)
+     * @param listener an object that will be invoked when events of type {@link org.jboss.tm.usertx.TransactionEvent}
+     *                 occur
+     * @throws NotSupportedException if transaction does not support the listener
+     * feature (or the TSR resource map key is in use
+     */
+    void addListener (Transaction transaction, TransactionListener listener) throws NotSupportedException;
```

Comment:
I think it could be useful to be able to flag an exception here in the future so maybe we should just define a new Exception type?

org.jboss.tm.TransactionException